### PR TITLE
[plugin.program.newgrounds] 1.0.5

### PR DESCRIPTION
**Pull request by erfg12**
### Add-on details:
- **General**
  - Add-on name: Newgrounds
  - Add-on ID: newgrounds
  - Version number: 1.0.5
  - Kodi/repository version: Leia
- **Code location**
  - URL: https://github.com/erfg12/kodi-repo/tree/master/plugin.program.newgrounds
  - Revision/tag: c7dee26a70606bea20d9d83efbb7b69db29f2385
  - Branch: Master
### Description of changes:
Changed cURL to JSON API

### CODE REVIEW

**Comment by Lunatixz** [comment]
Please rename PR to the proper format. [addon.id] version, and squash into a single commit. THX

**Comment by erfg12** [comment]
@Lunatixz fixed I think.

File: plugin.program.newgrounds/addon.py [hunk 1]
```diff
@@ -48,64 +49,67 @@ def build_url(query):
...
-    urllib2.install_opener(opener)
-    # Debug
+def getFeatured(content, mode, cType):
+    url = "https://newgrounds.app/featured.cache.php"
+    data = None
+
+    req = urllib2.Request(url, data, hdr)
+    response = urllib2.urlopen(req)
+    r = json.loads(response.read())
+    #xbmcgui.Dialog().ok('DEBUG', json.dumps(r))
```
Comment (basrieter) [inline plugin.program.newgrounds/addon.py]: Can you remove this commented debug code?

File: plugin.program.newgrounds/addon.py [hunk 2]
```diff
@@ -48,64 +49,67 @@ def build_url(query):
...
+        xbmcplugin.addDirectoryItem(handle=addon_handle, url=url, listitem=li, isFolder=True)
+    xbmcplugin.endOfDirectory(addon_handle)
+
+def searchContent(content, mode, cType):
+    if (int(args['page'][0]) > 1): #next page
+        search = args['search_term'][0]
+    else: #initial search
+        kb = xbmc.Keyboard('', 'search')
+        kb.doModal()
+        if kb.isConfirmed():
```
Comment (basrieter) [inline plugin.program.newgrounds/addon.py]: what if the user cancelled? Then the `search` variable does not exist and stuff breaks.

File: plugin.program.newgrounds/addon.py [hunk 3]
```diff
@@ -48,64 +49,67 @@ def build_url(query):
...
+    response = urllib2.urlopen(req)
+    r = json.loads(response.read())
+    for fm in r['browse_content']:
+        url = build_url({'mode': mode, 'contentID': fm['content_url']})
+        li = xbmcgui.ListItem(fm['title'], iconImage=fm['thumbnail_newgrounds_url'] + fm['thumbnail'])
+        li.setInfo(cType, {'title': fm['title']})
+        xbmcplugin.addDirectoryItem(handle=addon_handle, url=url, listitem=li, isFolder=True)
+
+    url = build_url({'mode': mode, 'page': nextPage, 'search_term': search})
+    li = xbmcgui.ListItem('NEXT PAGE >', iconImage='DefaultVideo.png')
```
Comment (basrieter) [inline plugin.program.newgrounds/addon.py]: Please localize the _NEXT PAGE >_ label.

File: plugin.program.newgrounds/addon.py [hunk 4]
```diff
@@ -48,64 +49,67 @@ def build_url(query):
...
-    return get_request(url, urllib.urlencode(values))
+    r = json.loads(response.read())
+    for fm in r['browse_content']:
+        url = build_url({'mode': mode, 'contentID': fm['content_url']})
+        li = xbmcgui.ListItem(fm['title'], iconImage=fm['thumbnail_newgrounds_url'] + fm['thumbnail'])
+        li.setInfo(cType, {'title': fm['title']})
+        xbmcplugin.addDirectoryItem(handle=addon_handle, url=url, listitem=li, isFolder=True)
+
+    url = build_url({'mode': mode, 'cat': args['cat'][0], 'page': nextPage})
+    li = xbmcgui.ListItem('NEXT PAGE >', iconImage='DefaultVideo.png')
```
Comment (basrieter) [inline plugin.program.newgrounds/addon.py]: Same here, localize _NEXT PAGE >_

File: plugin.program.newgrounds/addon.py [hunk 5]
```diff
@@ -115,13 +119,9 @@ def login ():
...
-    #if 'You have successfully signed in!' in login():
-    #    xbmcgui.Dialog().ok('DEBUG', 'logged in!')
+artCats = {"all", "illustration", "fine-art", "3d-art", "pixel-art", "other"}
 
-    #li = xbmcgui.ListItem('[B]-- AUDIO PORTAL --[/B]', iconImage='DefaultAudio.png')
-    #xbmcplugin.addDirectoryItem(handle=addon_handle, url='#', listitem=li, isFolder=False)
+if mode is None:
 
     url = build_url({'mode': 'featured_audio'})
     li = xbmcgui.ListItem('[COLOR orange]Featured Audio[/COLOR]', iconImage='DefaultAudio.png')
```
Comment (basrieter) [inline plugin.program.newgrounds/addon.py]: Please localize all these labels.

**Pull request closed without merging**